src/util/serializing/ObjectEncoding.cpp

```diff
@@ -1,13 +1,15 @@
 #include "util/serializing/ObjectEncoding.h"
 
-ObjectEncoding::ObjectEncoding() { this->data = g_string_new(""); }
+#include <string>
+
+ObjectEncoding::ObjectEncoding() = default;
 
 ObjectEncoding::~ObjectEncoding() = default;
 
-void ObjectEncoding::addStr(const char* str) const { g_string_append(this->data, str); }
+void ObjectEncoding::addStr(const char* str) { this->data.append(std::string(str)); }
 
-auto ObjectEncoding::getData() -> GString* {
-    GString* str = this->data;
-    this->data = nullptr;
-    return str;
+auto ObjectEncoding::getData() -> std::string {
+    std::string res = this->data;
+    this->data.resize(0);
+    return res;
```

Comment:
copy, and then reset is inefficient, move or exchange it.
```suggestion
    return std::move(this->data);
```

Reply:
Done